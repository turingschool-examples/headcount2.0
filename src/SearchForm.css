@import url(https://fonts.googleapis.com/css?family=Lora|Roboto:400,500);

.search-field {
  width: 100%;
  display: flex;
  justify-content: center;
}

input {
  width: 400px;
  padding: 10px;
  border-radius: 10px;
  margin-bottom: 20px;
  outline: none;
  font-size: 16px;
  font-family: 'Lora', sans-serif;
} 

