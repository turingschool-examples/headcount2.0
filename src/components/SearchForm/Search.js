import React, { Component } from 'react';
import PropTypes from 'prop-types';

export class Search extends Component {
  constructor() {
    super();
    this.state = {
      query: ''
    };
  }

  handleChange =  event => {
    const { value } = event.target;

    this.props.filterDistricts(value);

    this.setState({ query: value });
  }

  render() {
    const { query } = this.state;

    return (
      <input
        onChange={this.handleChange}
        type="text"
        name="query"
        value={query}
        placeholder="Search"
      />
    );
  }
}

Search.propTypes = {
  filterDistricts: PropTypes.func.isRequired
};

export default Search;
