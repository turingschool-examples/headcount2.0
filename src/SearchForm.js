import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './SearchForm.css'

class SearchForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      schoolName: ''
    }
  }

saveUserInput = (e) => {
  this.setState({
    schoolName: e.target.value
  })
  this.props.filterCards(this.state.schoolName)
}

render() { 
  return (
    <div>
      <form className="search-field">
        <input type="text" 
               placeholder="search for a school district..."
               onChange={(e) => {this.saveUserInput(e)}}
               />
      </form>
    </div>
   )
 }
}

SearchForm.propTypes = {
  filterCards: PropTypes.func
}


export default SearchForm; 