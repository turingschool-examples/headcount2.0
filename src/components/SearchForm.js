import React, { Component } from "react";
import PropTypes from "prop-types";
import "../css/Search.css";

class SearchForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      districtSearch: ""
    };
  }

  handleChange = event => {
    this.setState({
      districtSearch: event.target.value
    });
    this.props.findDistricts(event.target.value);
  };

  render() {
    return (
      <form onSubmit={event => event.preventDefault()}>
        <input
          className="search-input"
          type="text"
          placeholder="Search Districts Here"
          value={this.state.districtSearch}
          onChange={this.handleChange}
        />
      </form>
    );
  }
}

SearchForm.propTypes = {
  findDistricts: PropTypes.func
};

export default SearchForm;
