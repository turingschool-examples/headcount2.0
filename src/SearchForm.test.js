import React from 'react';
import { shallow } from 'enzyme';

import SearchForm from './SearchForm.js'

describe('SearchForm', () =>{

  it('should match snapshot', () => {
    const school = {filterCards: 'COLORADO'}
    const wrapper = shallow(<SearchForm filterCards={school}/>)

    expect(wrapper).toMatchSnapshot()
  })

  it('should save user input to state', () => {
    const mockFunc = jest.fn()
    const wrapper = shallow(<SearchForm filterCards={mockFunc}/>)

    const mockEvent = {target: {value: 'COLORADO'}}
    wrapper.find('input').simulate('change', mockEvent)

    wrapper.instance().saveUserInput(mockEvent)
    expect(wrapper.state()).toEqual({"schoolName": "COLORADO"})

  })

})