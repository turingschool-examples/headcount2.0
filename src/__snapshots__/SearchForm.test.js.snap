// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match snapshot 1`] = `
<div>
  <form
    className="search-field"
  >
    <input
      onChange={[Function]}
      placeholder="search for a school district..."
      type="text"
    />
  </form>
</div>
`;
